src/SecureHeadersHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SecureHeaders\PsrHttpAdapter;
+
+use Aidantwoods\SecureHeaders\SecureHeaders;
+use Psr\Http\Message\RequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+
+/**
+ * Secure headers handler
+ *
+ * Middleware to apply Secure headers to a PSR7 response
+ */
+class SecureHeadersHandler
+{
+    /**
+     * Headers
+     *
+     * @var SecureHeaders
+     *
+     * @access protected
+     */
+    protected $headers;
+
+    /**
+     * __construct
+     *
+     * @param SecureHeaders $headers Configured headers instance
+     *
+     * @access public
+     */
+    public function __construct(SecureHeaders $headers)
+    {
+        $this->headers = $headers;
+    }
+
+    /**
+     * Handle PSR7 Request
+     *
+     * Delegates to middleware chain and applies secure headers to the returned
+     * response object before returning it
+     *
+     * @param Request  $request  Incoming PSR7 request
+     * @param Response $response PSR7 Response
+     * @param callable $next     Delagate middleware
+     *
+     * @return Response
+     *
+     * @access public
+     */
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+        $response = $next($request, $response);
+        $headers  = $this->headers;
+        $adapter  = $this->adapt($response);
```

Comment:
Is this needed, as opposed to just `$adapter = new Psr7Adapter($response)`? 😉

Reply:
I did this to try and minimize the responsibility of `__invoke`, and have the responsibility of instantiation contained within a single method that only has that responsibility. I was tempted to do one of these:

```php


class ApplySecureHeaders
{

    protected $headers;

    protected $adapterClass;

    public function __construct(
        SecureHeaders $headers,
        $adapterClass =Psr7Adapter::class
    ) {
        $this->headers = $headers;
        $this->adapterClass = $adapterClass;
    }

    //...

    protected function adapt(Response $response)
    {
        $class = $this->adapterClass;
        $return new $class($response);
    }
}

/**
 *
 *   OR something like
 *
 **/

class ApplySecureHeaders
{

    protected $headers;

    protected $factory;

    public function __construct(
        SecureHeaders $headers,
        AdapterFactoryInterface $AdapterFactory = null
    ) {
        $this->headers = $headers;
        $this->AdapterFactory = $adapterFactory ?? new AdapterFactory;
    }

    public function __invoke(Request $request, Response $response, callable $next)
    {
        $response = $next($request, $response);
        $headers  = $this->headers;
        $adapter  = $this->adapterFactory->newInstance($response);

        $headers->apply($adapter);
        $response = $adapter->getFinalResponse();

        return $response;
    }
}
```

... so this was my way of not using a proper factory, but still providing the ability to override the adapter with minimal effort. This way you can extend the handler, and override the adapt method to return a different implementation of `Adapter` if one needed to. 

Let me know if you think it should change.

Thanks!

Reply:
> I did this to try and minimize the responsibility of `__invoke`, and have the responsibility of instantiation contained within a single method that only has that responsibility.

Alright, that seems like a sensible goal :D we'll keep it as